src/server/mod.rs

```diff
@@ -1 +1,279 @@
+                net::send(&mut stream, MessageType::TurnHost);
+
+                // wait for input from Host
+                println!("{}", Yellow.paint("It's your turn!"));
+                let mut coord;
+                loop {
+                    println!("{}", Yellow.paint("Please enter a valid coordinate: "));
+                    coord = util::read_string();
+                    if ::model::valid_coordinate(&coord) {
+                        break;
+                    }
+                    print!("{}", Red.paint("Invalid coordinate! "));
+                }
+                // modify boards
+                let coord_id = Board::get_index(&coord);
+                match ::model::match_move(&mut host, &mut client, coord_id) {
+                    SubField::Hit => {
+                        net::send(&mut stream, MessageType::Hit(coord_id));
+                        model::print_boards(&host);
+                    }
+                    SubField::Miss => {
+                        net::send(&mut stream, MessageType::Miss(coord_id));
+                        model::print_boards(&host);
+                        current_player = CurrentPlayer::Client;
+                    }
+                    _ => {}
+                }
+
+                // if Host won: send message to Client, end game
+                if ::model::game_over(&client) {
+                    net::send(&mut stream, MessageType::Lost);
+                    println!("{}", Yellow.paint("Congratulations, you won the game :)"));
+                    break;
+                }
+            }
+            CurrentPlayer::Client => {
+                println!(
+                    "{} {} {}",
+                    Cyan.paint("Wait for"),
+                    Yellow.paint(&client.name),
+                    Cyan.paint("to finish turn!"),
+                );
+                // inform Client that its his turn
+                net::send(&mut stream, MessageType::RequestCoord);
+                // wait for input from Client
+                let recv: Result<types::MessageType, DeserializeError> =
+                    deserialize_from(&mut stream, bincode::SizeLimit::Infinite);
+                let coordinate = match recv {
+                    Ok(received) => {
+                        match received {
+                            MessageType::Shoot(coord) => {
+                                coord
+                            },
+                            MessageType::Quit => {
+                                println!("Client closed connection.");
+                                return
+                            },
+                            _ => {
+                                // unexpected packet
+                                "".to_string()
+                            },
+                        }
+                    },
+                    Err(_) => {
+                        println!("ERROR coord");
+                        return
+                    },
```

Comment:
see above

Thinking about it: it would be nice to somehow... abstract this match block. I've seen a match block like this a few times already. I think wrapping it into a function is not yet feasible ([this](https://doc.rust-lang.org/std/mem/fn.discriminant.html) would  be needed). But you could maybe write a macro for it. 

It's always the same: you expect one specific package. You also accept `Quit` packages, but you want to error/panic on all other packages. Maybe a macro would be nice.